src/animation/AnimatedComponent.js

```diff
@@ -1,53 +1,58 @@
 /* @flow strict-local */
-import React, { PureComponent } from 'react';
+import React, { useRef, useCallback, useLayoutEffect } from 'react';
 import type { Node as React$Node } from 'react';
 import { Animated, Easing } from 'react-native';
 
+import { usePrevious } from '../reactUtils';
 import type { Style } from '../types';
 
 type Props = $ReadOnly<{|
   stylePropertyName: string,
   fullValue: number,
   children: React$Node,
   style?: Style,
-  visible: boolean,
-  useNativeDriver: boolean,
-  delay: number,
+  visible?: boolean,
+  useNativeDriver?: boolean,
+  delay?: number,
 |}>;
 
-export default class AnimatedComponent extends PureComponent<Props> {
-  static defaultProps = {
-    visible: true,
-    useNativeDriver: true,
-    delay: 0,
-  };
-
-  animatedValue = new Animated.Value(0);
-
-  animate() {
-    Animated.timing(this.animatedValue, {
-      toValue: this.props.visible ? this.props.fullValue : 0,
-      delay: this.props.delay,
+/**
+ * Animates the specified style property on visibility change.
+ */
+export default function AnimatedComponent(props: Props): React$Node {
+  const {
+    visible = true,
+    useNativeDriver = true,
+    delay = 0,
+    fullValue,
+    children,
+    stylePropertyName,
+    style,
+  } = props;
+
+  const prevVisible = usePrevious(visible);
+
+  const animatedValue = useRef(new Animated.Value(0));
```

Comment:
Hmm, yeah, interesting...

Reply:
See "You might also occasionally want to avoid re-creating the useRef() initial value" [here](https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily).

Reply:
Cool, yeah. That looks like something we'd want to do if we actually know the constructor is expensive, but not something we'd want to default to speculatively.